src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManager.cs

```diff
@@ -226,6 +226,11 @@ private static OpenType GetOpenType(Type t)
                 }
             }
 
+            if (typeDest == typeof(JArray) && typeof(IEnumerable).IsAssignableFrom(typeSource))
```

Comment:
>typeof(IEnumerable) [](start = 46, length = 19)

This may be too general. See checks at line 290 below.  For example, this may pull in System.String. 
For now, we should restrict this to IEnumebale<T> where T = JObject or JToken.  We can always make more general later.